config_splunk.go

```diff
@@ -0,0 +1,66 @@
+package evergreen
+
+import (
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type SplunkConfig struct {
+	SplunkConnectionInfo send.SplunkConnectionInfo `bson:",inline" json:",inline" yaml:",inline"`
+}
+
+func (c *SplunkConfig) SectionId() string { return "splunk" }
+
+func (c *SplunkConfig) Get(env Environment) error {
+	ctx, cancel := env.Context()
+	defer cancel()
+	coll := env.DB().Collection(ConfigCollection)
+
+	res := coll.FindOne(ctx, byId(c.SectionId()))
+	if err := res.Err(); err != nil {
+		if err == mongo.ErrNoDocuments {
+			// Get the splunk config from global if the document doesn't exist.
```

Comment:
Is this temporary code? Should we Set it in this case so that we can remove this code after staging/prod have been updated to set it in a separate document?

Reply:
this would only be called after a change to the admin page so i think this might need to be semi permanent

Reply:
Yeah I am thinking we should do something to the admin page to update it though so that we can remove this code. If it doesn't need to be something in the Splunk section we could just add / remove a dummy username to one of the host configs (like GCE) or we could add/remove something from the Splunk channel section.

Reply:
https://jira.mongodb.org/browse/EVG-17353